src/unsafe/calling-unsafe-functions.md

```diff
@@ -1,11 +1,14 @@
 # Calling Unsafe Functions
 
 A function or method can be marked `unsafe` if it has extra preconditions you
-must uphold:
+must uphold to avoid undefined behaviour:
 
 ```rust,editable
 fn main() {
     let emojis = "🗻∈🌏";
+
+    // Safe because the indices are in the correct order, within the bounds of
```

Comment:
This duplicates the comment just below. I made the comment short on purpose to make things fit on the screen in the classroom.

Reply:
Yeah, please remove the comment and preserve the vertical height.

Reply:
I think this comment would imply that _the indices are in the correct order_ and _within the bounds of the string slice_:

```suggestion
    // Safe because the indices lie on UTF-8 sequence boundaries.
```

It's just the opposite of the comment below, so it has the same size :-)

Reply:
That's not enough for the call to be safe though. I really want to drill into people that if you're calling an unsafe function, you need to read all the safety requirements documented for that function and ensure that they are all met, and explain that in your safety comment.

I keep seeing in code reviews of unsafe Rust cases of people not doing that, and writing unsound code as a result, so I think it's important to get people thinking carefully and precisely about unsafe code right from the start, as soon as we introduce it. If we're going to cover unsafe Rust at all, we should make sure people know how to do it properly, otherwise we're going to end up with lots of sloppy unsafe Rust in our codebase and lose a large part of the benefit of using Rust in the first place.